boards: silabs: Add xg27 radio boards

asmellby opened the pull request:
Reformat xg27 soc devicetree files according to https://docs.zephyrproject.org/latest/contribute/style/devicetree.html and https://docs.kernel.org/devicetree/bindings/dts-coding-style.html:

* Sort nodes by unit address or name
* Sort properties by category and name

Add support for all socs in the xg27 family.

Add three xg27 radio boards: `bg27_rb4110b`, `bg27_rb4111b` and `xg27_rb4194a`.

asmellby committed:
dts: arm: silabs: xg27: Reformat dtsi files

Reformat devicetree files:
 * Sort nodes by unit address or name
 * Sort properties by category and name

Add missing properties to existing nodes.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
dts: arm: silabs: xg27: Add missing nodes

Add missing peripheral and clock nodes for xg27.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
soc: silabs: Add complete xg27 soc family

Add efr32bg27 and efr32mg27 socs.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
drivers: entropy: gecko: Configure for all xg27 socs

Use cryptoacc variant on all xg27 socs.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
boards: silabs: xg27_dk2602a: Set cells properties on usart0

Set #address-cells and #size-cells on usart0. Since usarts can
be used for both UART and SPI, this property shouldn't be set
in SoC-level DTS, since it makes it harder to switch compatible.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

asmellby committed:
boards: silabs: Add xg27 radio boards

Add 3 xg27 radio boards:

* bg27_rb4110b
* bg27_rb4111b
* xg27_rb4194a

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

jerome-pouiller commented on boards/silabs/radio_boards/xg27/bg27_rb411xb.dtsi:
What is the risk to have a future rb4119b that does not match with this definition? That's said, I have not better proposal.

asmellby commented on boards/silabs/radio_boards/xg27/board.cmake:
```diff
@@ -0,0 +1,11 @@
+board_runner_args(jlink "--device=EFR32MG27CxxxF768")
+else()
+board_runner_args(jlink "--device=EFR32BG27CxxxF768")
+endif()
```
Won't have `xxx` in the middle...

Martinhoff-maker commented on soc/silabs/silabs_s2/xg27/Kconfig.soc:
```diff
@@ -13,12 +13,64 @@ config SOC_SERIES_EFR32BG27
+	default "efr32bg27c230f768im32" if SOC_EFR32BG27C230F768IM32
+	default "efr32bg27c230f768im40" if SOC_EFR32BG27C230F768IM40
+	default "efr32bg27c320f768gj39" if SOC_EFR32BG27C320F768GJ39
+	default "efr32bg27c320f768ij39" if SOC_EFR32BG27C320F768IJ39
```
I think you miss the dtsi for this one ?

jhedberg reviewed (dismissed):
Looks good. Please update `west.yml` now that the corresponding module PR was merged.

asmellby commented on boards/silabs/radio_boards/xg27/bg27_rb411xb.dtsi:
```diff
@@ -0,0 +1,250 @@
+		compatible = "gpio-keys";
+
+		button0: button_0 {
+			gpios = <&gpiob 2 GPIO_ACTIVE_LOW>;
```
They are on the same pin: 
<img width="891" height="120" alt="image" src="https://github.com/user-attachments/assets/57634d7d-dc90-467f-a5ee-a43c656738c3" />

asmellby replied:
Hm. Disabling the LED node in board dts will prevent the blinky sample from working, so that's not a great solution. Maybe just removing the `led0` alias in a board overlay for the button sample is the most sensible approach? Could also swap 0 and 1 around, but then the dts nodes don't match the silkscreen on the board...

asmellby committed:
samples: basic: button: Disable led on bg27 boards

Button 0 and LED 0 share a GPIO pin on bg27 boards. Remove the
`led0` alias in a board overlay to make the button sample work
out of the box.

Signed-off-by: Aksel Skauge Mellbye <aksel.mellbye@silabs.com>

nordicjm reviewed (commented):
rebase is needed

kartben reviewed (approved):
there are probably a few other samples that might have incorrect out-of-the-box experience due to button/led overlap but this is probably fine given that the caveat is called out in the board's documentation
